readme.md

```diff
@@ -19,45 +19,78 @@ If there is an API endpoint that isn't currently supported, feel free to open an
-```
+`Install-Package Forem.Api -Version 0.4.0`	
 
-```csharp
-using Forem.Api;
+## Usage
 
-public class MyClass
-{
-	private IArticlesService ArticlesService { get; }
-	private ITagsService TagsService { get; }
+- For Some of the methods, DEV API key `apiKey`  is required to be passed to authenticate you as a user. 
+  To Obtain one, check the authentication section [GET API KEY](https://docs.dev.to/api/#section/Authentication)
 
-	public MyClass(IArticlesService articlesService, ITagsService tagsService)
-	{
-		ArticlesService = articlesService;
-		TagsService = tagsService;
-	}
+- Using DI, Add the service to the container just by doing the below:
+  `services.AddForemApi(new Uri("https://dev.to/"));`
 
-	public async Task DoWork()
+- Configure your `HttpClient` DI by adding the below snippet to your `startup.cs` file (or wherever you're configuring your DI things):
+
+```csharp
+	services.AddHttpClient();
+	services.AddTransient(provider =>
 	{
-		var articles = await ArticlesService.GetArticlesAsync();
-		
-		// Your code here...
-	}
-}
+		return provider.GetRequiredService<IHttpClientFactory>().CreateClient(string.Empty);
+	});
+
 ```
 
-### Without DI
+Havng all this setup, then you're good to go!!
+
+
+## Snippets
+
+### Articles
 
 ```csharp
-using Forem.Api;
-using System.Net.Http;
+    using Forem.Api;
+
+	
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ArticleController : ControllerBase
+    {
+        private readonly IArticlesService _articlesService;
+        private readonly ITagsService _tagsService;
+
+        //Without Dependency Injection - global
+        ArticlesService articlesService = new ArticlesService(new Uri("https://dev.to/"), new HttpClient());
+
+        //With Dependency Injection
+        public ArticleController(IArticlesService articlesService, ITagsService tagsService)
+        {
+            _articlesService = articlesService;
+            _tagsService = tagsService;
+        }
+
+```
 
-var articlesService = new ArticlesService(new Uri("https://dev.to/"), new HttpClient());
-var tagsService = new TagsService(new Uri("https://dev.to/"), new HttpClient());
+To get all articles you simply just call `GetArticlesAsync` passing optional params `page` and `perPage` with default values `1` & `30` respectively.
+
+```csharp
+	[HttpGet("all")]
+        public async Task<IActionResult> GetAllArticles(int page, int itemPerPage)
+        {
+            //Without DI - local
+            //var articlesService = new ArticlesService(new Uri("https://dev.to/"), new HttpClient());
+           
+            var articles = await articlesService.GetArticlesAsync();
+
+            articles = await _articlesService.GetArticlesAsync(page, itemPerPage);
+            
+            return Ok(articles);
+        }
 
-var articles = await articlesService.GetArticlesAsync();
 ```
+
+Feel free to check out other methods/endpoints [Aticles Doc](https://docs.dev.to/api/#operation/getArticles)
```

Comment:
Nit: Typo in "Articles", missing the "r"

Reply:
I will make the corrections. Thanks